src/measure_theory/borel_space.lean

```diff
@@ -434,8 +470,62 @@ lemma measurable.sub [add_group α] [topological_add_group α] [second_countable
   measurable (λ x, f x - g x) :=
 hf.add hg.neg
 
-lemma measurable.is_lub [linear_order α] [order_topology α] [second_countable_topology α]
-  {ι} [encodable ι] {f : ι → δ → α} {g : δ → α} (hf : ∀ i, measurable (f i))
+lemma measurable_comp_iff_of_closed_embedding {f : δ → β} (g : β → γ) (hg : closed_embedding g) :
+  measurable (g ∘ f) ↔ measurable f :=
+begin
+  refine ⟨λ hf, _, λ hf, hg.continuous.measurable.comp hf⟩,
+  apply measurable_of_is_closed, intros s hs,
+  convert hf (hg.is_closed_map s hs).is_measurable,
+  rw [@preimage_comp _ _ _ f g, preimage_image_eq _ hg.to_embedding.inj]
+end
+
+section linear_order
+
+variables [linear_order α] [order_topology α] [second_countable_topology α]
+
+lemma measurable_of_Iio {f : δ → α} (hf : ∀ x, is_measurable (f ⁻¹' Iio x)) : measurable f :=
+begin
+  convert measurable_generate_from _,
+  exact borel_space.measurable_eq.trans (borel_eq_generate_Iio _),
+  rintro _ ⟨x, rfl⟩, exact hf x
+end
+
+lemma measurable_of_Ioi {f : δ → α} (hf : ∀ x, is_measurable (f ⁻¹' Ioi x)) : measurable f :=
+begin
+  convert measurable_generate_from _,
+  exact borel_space.measurable_eq.trans (borel_eq_generate_Ioi _),
+  rintro _ ⟨x, rfl⟩, exact hf x
+end
+
+lemma measurable_of_Iic {f : δ → α} (hf : ∀ x, is_measurable (f ⁻¹' Iic x)) : measurable f :=
+begin
+  apply measurable_of_Ioi,
+  simp_rw [← compl_Iic, preimage_compl, is_measurable.compl_iff],
+  assumption
+end
+
+lemma measurable_of_Ici {f : δ → α} (hf : ∀ x, is_measurable (f ⁻¹' Ici x)) : measurable f :=
+begin
+  apply measurable_of_Iio,
+  simp_rw [← compl_Ici, preimage_compl, is_measurable.compl_iff],
+  assumption
+end
+
+lemma measurable_of_is_measurable_lt {f : δ → α} (hf : ∀ y, is_measurable {x | f x < y}) :
+  measurable f :=
+begin
+  rw [‹borel_space α›.measurable_eq, borel_eq_generate_Iio],
+  apply measurable_generate_from, rintro _ ⟨y, rfl⟩, exact hf y
+end
```

Comment:
Seems to be defeq to
```suggestion
measurable_of_Iio hf
```

Reply:
You are right, I didn't need both. I removed the `measurable_of_is_measurable_l*` versions.